src/firefly/js/ui/InputAreaFieldView.jsx

```diff
@@ -0,0 +1,151 @@
+}
+
+const ICON_SPACE_STYLE= {
+    verticalAlign: 'middle',
+    paddingLeft: 3,
+    width: 16,
+    height: 16,
+    display:'inline-block'};
+
+
+export class InputAreaFieldView extends Component {
+    constructor(props) {
+        super(props);
+        this.warnIcon = null;
+        this.state = { hasFocus: false, infoPopup: false };
+    }
+
+    shouldComponentUpdate(np, ns) { return sCompare(this, np, ns); }
+
+
+    componentDidUpdate() {
+        var {infoPopup}= this.state;
+        if (infoPopup) {
+            var {warningOffsetX, warningOffsetY}= computeWarningXY(this.warnIcon);
+            var {message}= this.props;
+            this.hider = DialogRootContainer.showTmpPopup(makeInfoPopup(message, warningOffsetX, warningOffsetY));
+        }
+        else {
+            if (this.hider) {
+                this.hider();
+                this.hider = null;
+            }
+        }
+    }
+
+    makeWarningArea(warn) {
+        if (warn) {
+            return (
+                <div style={ICON_SPACE_STYLE}
+                     onMouseOver={() => this.setState({infoPopup:true})}
+                     onMouseLeave={() => this.setState({infoPopup:false})}>
+                    <img src={EXCLAMATION} ref={(c) => this.warnIcon= c}/>
+                </div>
+            );
+        }
+        else {
+            return <div style={ICON_SPACE_STYLE}/>;
+        }
+    }
+
+    render() {
+        var {hasFocus}= this.state;
+        var {visible,label,tooltip,rows,cols,labelWidth,value,style,wrapperStyle,
+             valid,size,onChange, onBlur, onKeyPress, showWarning, message, type}= this.props;
+        if (!visible) return null;
+        wrapperStyle = Object.assign({whiteSpace:'nowrap', display: this.props.inline?'inline-block':'block'}, wrapperStyle);
+        return (
+            <div style={wrapperStyle}>
+                {label && <InputFieldLabel label={label} tooltip={tooltip} labelWidth={labelWidth}/> }
+                <textarea style={Object.assign({display:'inline-block'}, style)}
+                          rows={rows}
+                          cols={cols}
+                          className={computeStyle(valid,hasFocus)}
+                       onChange={(ev) => onChange ? onChange(ev) : null}
+                       onFocus={ () => !hasFocus ? this.setState({hasFocus:true, infoPopup:false}) : ''}
+                       onBlur={ (ev) => {
+                                onBlur && onBlur(ev);
+                                this.setState({hasFocus:false, infoPopup:false});
+                            }}
+                       onKeyPress={(ev) => onKeyPress && onKeyPress(ev)}
+                       value={type==='file' ? undefined : value}
+                       title={ (!showWarning && !valid) ? message : tooltip}
+                       size={size}
+                       type={type}
+                />
+                {showWarning && this.makeWarningArea(!valid)}
+            </div>
+        );
+    }
+}
```

Comment:
So much of the code is the same as `InputFieldView` I wonder it we should look at how to share the code between the two.  Also, I never liked the way we show errors for GWT in the area view.  It never looked quite right.  We might want to consider that now.